tools/includes/version-compare.sh

```diff
@@ -88,3 +88,104 @@ function version_compare {
 		[[ ${#A[@]} -gt ${#B[@]} ]]
 	fi
 }
+
+# Compute the difference between two version numbers, semver style.
+#
+# Note this is a bit looser than `pnpm semver`, as it accepts 4-part versions.
+#
+# @param $1 First version.
+# @param $2 Second version.
+# @return 0 if equal, 1 if suffix, 2 if patch, 3 if minor, 4 if major.
+function version_diff {
+	# Variable for the version_difference call
+	local INCREMENT
+
+	# Returning 0 if neither $1 > $2, nor $2 > $1
+	if ! version_compare "$1" "$2" 1 && ! version_compare "$2" "$1" 1; then
+		return 0
+	fi
+
+	if version_compare $1 $2; then
+
+		# $1 is greater than $2
+		version_increment $2 $1
+	else
+
+		# $2 is greater than $1
+		version_increment $1 $2
+	fi
+
+	echo "$INCREMENT"
+}
+
+# Compute the increment needed to get from $1 to $2, semver style.
+#
+# Note this is a bit looser than `pnpm semver`, as it accepts 4-part versions.
+#
+# @param $1 First version.
+# @param $2 Second version.
+function version_increment {
+
+	local V1="${1%%+*}" V2="${2%%+*}"
+
+	local A=() B=() i DIFF
+
+	# First, compare the version parts.
+	IFS='.' read -r -a A <<<"${V1%%-*}"
+	IFS='.' read -r -a B <<<"${V2%%-*}"
+
+	while [[ ${#A[@]} -lt ${#B[@]} ]]; do
+		A+=( 0 )
+	done
+	while [[ ${#B[@]} -lt ${#A[@]} ]]; do
+		B+=( 0 )
+	done
+
+	i=0
+
+	# If AA is ahead of BB, we have found our difference.
+	while [[ $i -lt ${#A[@]} && $i -lt ${#B[@]} ]]; do
+		local AA=${A[$i]}
+		local BB=${B[$i]}
+
+		if [[ $AA -lt $BB ]]; then
+			let DIFF="4 - $i"
+
+			INCREMENT=$DIFF
+			return
+		fi
+		i=$((i + 1))
+	done
+
+	# Version parts were equal, this means that the difference is the suffix.
+	INCREMENT=1
+}
+
+# Returns true if the version is patch level, semver style. This function disregards string suffixes,
+# so that 3.4.0-alpha would be considered as 3.4.0.
+#
+# @param $1 Version to check.
+# @return 1 if no patch level, 0 if patch level.
+function version_is_patch {
+	local V1="${1%%+*}"
+
+	local A=()
+
+	# First, compare the version parts.
+	IFS='.' read -r -a A <<<"${V1%%-*}"
+
+	# Checking if it's a four point style release
+	PATCH=${A[3]}
+
+	# If it exists, it's a patch level version
+	[ -n "$PATCH" ] && return 0
+
+	# Getting the third element of the array
+	PATCH=${A[2]}
+
+	[ -z "$PATCH" ] && return 1
+
+	[[ $PATCH -eq "0" ]] && return 1
+
+	return 0
```

Comment:
Since the return value of the function is by default the return value of the last command, you could just do like
```suggestion
	[[ $PATCH -ne "0" ]]
```